pulp_container/app/migrations/0020_update_push_repo_perms.py

```diff
@@ -0,0 +1,69 @@
+# Generated by Django 2.2.23 on 2021-06-01 14:53
+
+from django.contrib.auth.models import Group
+from django.db import migrations
+from guardian.shortcuts import assign_perm, get_objects_for_group
+
+
+def update_repo_perms(apps, schema_editor):
+    """ Assign push repo perms to the namespace and distribution groups. """
+    # Group = apps.get_model("auth", "Group")
+    AccessPolicy = apps.get_model('core', 'AccessPolicy')
+    push_repo_vewset = 'repositories/container/container-push'
+    dist_viewset = 'distributions/container/container'
+    namespace_vewset = 'pulp_container/namespaces'
+    viewset_names = (dist_viewset, namespace_vewset, push_repo_vewset)
+    existing_perms = (
+        'container.modify_content_containerpushrepository',
+        'container.namespace_modify_content_containerpushrepository',
+    )
+    groups_ns = Group.objects.filter(
+        name__regex=r'container.(distribution|namespace).(collaborators|owners)'
+    )
+
+    for viewset_name in viewset_names:
+        try:
+            db_access_policy = AccessPolicy.objects.get(viewset_name=viewset_name)
+        except AccessPolicy.DoesNotExist:
+            pass
+        else:
+            if not db_access_policy.customized:
+                for perm in existing_perms:
+                    for group in groups_ns:
+                        objs = get_objects_for_group(group, perm, accept_global_perms=False)
+                        for obj in objs:
+                            if obj.ACCESS_POLICY_VIEWSET_NAME == namespace_vewset:
+                                assign_perm('container.namespace_change_containerpushrepository', group, obj)
+                            elif obj.ACCESS_POLICY_VIEWSET_NAME == push_repo_vewset:
+                                assign_perm('container.change_containerpushrepository', group, obj)
+
+
+def add_new_perms(apps, schema_editor):
+    """ Add new permssion to the table. """
+    Permission = apps.get_model('auth', 'Permission')
```

Comment:
I thought, we should only automatically touch existing permissions, if the default access_policy is in place unmodified.

Reply:
Oh, this is permissions. I thought the `AlterModelOptions` should handle this.

Reply:
This part is handled by signals in the post migrate operation and this is the part which took a lot of time to figure out :/
Emitting post migrate early doesn't sound safe, so a s a workaround we are creating/updating permissions manually to be able to assign them as a part of a migration.

Reply:
I see. Can we leave a comment here?

Reply:
yes, will do